src/typing/Typing.mli

```diff
@@ -9,3 +9,6 @@ val resolved_type_of_id_info :
   Lang.t -> AST_generic.id_info -> AST_generic.name Type.t
 
 val check_program : Lang.t -> AST_generic.program -> unit
+
+val guess_type_of_dotaccess :
```

Comment:
would be nice to add a comment saying this is used in the pro engine, which is why it's exposed, and also
maybe a short description of the function.